libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -96,32 +169,94 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
-/* ======================  TESTING C_Initialize  ============================ */
+/* ==========================  Helper functions  ============================ */
 
 /*!
- * @brief C_Initialize happy path.
+ * @brief Helper function to initialize PKCS #11.
  *
  */
-void test_pkcs11_C_Initialize( void )
+static CK_RV prvInitializePkcs11( SemaphoreHandle_t pxPKCS11Mutex )
 {
     CK_RV xResult = CKR_OK;
 
-    /* Hack: Giving out pointer to xResult when creating a mutex. This unit testcase
-     * will never use the actual mutex. */
-    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
+    xQueueCreateMutex_IgnoreAndReturn( pxPKCS11Mutex );
```

Comment:
If the test paths will never use the mutex, why create it?

Reply:
Hmm I did it this way so I could pass in a pointer but returning anything other than NULL should be sufficient, I will update this based on your suggestion.

Reply:
Done